src/Compilers/Core/Portable/SourceGeneration/Nodes/ValueSourceExtensions.cs

```diff
@@ -13,38 +13,38 @@ namespace Microsoft.CodeAnalysis
     public static class IncrementalValueProviderExtensions
     {
         // 1 => 1 transform 
-        public static IncrementalValueProvider<TResult> Select<TSource, TResult>(this IncrementalValueProvider<TSource> source, Func<TSource, CancellationToken, TResult> selector) => new IncrementalValueProvider<TResult>(new TransformNode<TSource, TResult>(source.Node, selector.WrapUserFunction()));
+        public static IncrementalValueProvider<TResult> Select<TSource, TResult>(this IncrementalValueProvider<TSource> source, Func<TSource, CancellationToken, TResult> selector) => new IncrementalValueProvider<TResult>(new TransformNode<TSource, TResult>(source.Node, selector.WrapUserFunction(source.CatchAnalyzerExceptions)), source.CatchAnalyzerExceptions);
```

Comment:
This feels incredible viral for such an edge-case change. Obviously, we need to thread it through somehow, but given this is a feature flag that is expected to be really a debugging only aid, could we instead just have a static field on `UserFunctionExtensions` that we just set in the ctor of the GeneratorDriver? 

I know its maybe not 'perfect' code, but it seems much less invasive for such a niche case

Reply:
I guess I would be fine with that, the question is: would I get a sign-off from @333fred on that, as well?

Reply:
I feel like you never know when the next time we're going to want to plumb more options through the generator driver is going to be. If anything, what I would maybe do is pull the bool into a struct so that we can add more options later :).

I don't _hate_ the static state (we do that with `TestOnlyCompilationData`, for example), but I'm not a fan either.

Reply:
> I feel like you never know when the next time we're going to want to plumb more options through the generator driver is going to be.

👍 

> pull the bool into a struct so that we can add more options later

Reasonable, but I think that can be done when needed in the future.